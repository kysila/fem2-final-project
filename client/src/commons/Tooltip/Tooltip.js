import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@material-ui/core/Tooltip';
import { useArrowStyles } from './styles';

function ArrowTooltip(props) {
  const { arrow, ...classes } = useArrowStyles();
  const [arrowRef, setArrowRef] = useState(null);

  return (
    <Tooltip
      open={Boolean(props.title)}
      classes={classes}
      PopperProps={{
        popperOptions: {
          modifiers: {
            arrow: {
              enabled: Boolean(arrowRef),
              element: arrowRef,
            },
          },
        },
      }}
      {...props}
      title={(
        <React.Fragment>
          {props.title}
          <span className={arrow} ref={setArrowRef} />
        </React.Fragment>
      )}
    />
  );
}

ArrowTooltip.propTypes = {
  title: PropTypes.node,
};

ArrowTooltip.defaultProps = {
  title: null,
};

export {
  ArrowTooltip,
};
