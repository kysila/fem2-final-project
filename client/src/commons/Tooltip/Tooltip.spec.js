/* eslint-disable */
import React from 'react';
import {shallow} from 'enzyme';
import { ArrowTooltip } from './Tooltip';

describe('Steps component', () => {
  it('Steps render empty title', () => {
    const rendered = shallow(
      <ArrowTooltip />,
    );

    expect(rendered).toMatchSnapshot();
  });

  it('Steps render filled title', () => {
    const rendered = shallow(
      <ArrowTooltip title="Some text here."/>,
    );

    expect(rendered).toMatchSnapshot();
  });
});
