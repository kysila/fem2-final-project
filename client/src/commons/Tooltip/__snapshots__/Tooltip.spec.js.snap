// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Steps component Steps render empty title 1`] = `
<WithStyles(ForwardRef(Tooltip))
  PopperProps={
    Object {
      "popperOptions": Object {
        "modifiers": Object {
          "arrow": Object {
            "element": null,
            "enabled": false,
          },
        },
      },
    }
  }
  classes={
    Object {
      "popper": "makeStyles-popper-3",
      "tooltip": "makeStyles-tooltip-1",
    }
  }
  open={false}
  title={
    <React.Fragment>
      <span
        className="makeStyles-arrow-2"
      />
    </React.Fragment>
  }
/>
`;

exports[`Steps component Steps render filled title 1`] = `
<WithStyles(ForwardRef(Tooltip))
  PopperProps={
    Object {
      "popperOptions": Object {
        "modifiers": Object {
          "arrow": Object {
            "element": null,
            "enabled": false,
          },
        },
      },
    }
  }
  classes={
    Object {
      "popper": "makeStyles-popper-3",
      "tooltip": "makeStyles-tooltip-1",
    }
  }
  open={true}
  title={
    <React.Fragment>
      Some text here.
      <span
        className="makeStyles-arrow-2"
      />
    </React.Fragment>
  }
/>
`;
